examples/stable-diffusion/training/train_text_to_image_sdxl.py

```diff
@@ -933,95 +954,58 @@ def preprocess_train(examples):
-            tokenizers=tokenizers,
-            proportion_empty_prompts=args.proportion_empty_prompts,
-            caption_column=args.caption_column,
-        )
-        compute_vae_encodings_fn = functools.partial(compute_vae_encodings, vae=vae)
-        with accelerator.main_process_first():
-            from datasets.fingerprint import Hasher
-
-            # fingerprint used by the cache for the other processes to load the result
-            # details: https://github.com/huggingface/diffusers/pull/4038#discussion_r1266078401
-            new_fingerprint = Hasher.hash(args)
-            new_fingerprint_for_vae = Hasher.hash("vae")
-
-            train_dataset = train_dataset.map(compute_embeddings_fn, batched=True, new_fingerprint=new_fingerprint)
-            train_dataset = train_dataset.map(
-                compute_vae_encodings_fn,
-                batched=True,
-                batch_size=args.train_batch_size * accelerator.num_processes * args.gradient_accumulation_steps,
-                new_fingerprint=new_fingerprint_for_vae,
-            )
+        train_dataset = dataset["train"]
+        if len(args.mediapipe) == 0:
+            # Set the training transforms
+            train_dataset = train_dataset.with_transform(preprocess_train)
+
+    compute_embeddings_fn = functools.partial(
+        encode_prompt,
+        text_encoders=text_encoders,
+        tokenizers=tokenizers,
+        proportion_empty_prompts=args.proportion_empty_prompts,
+        caption_column=args.caption_column,
+    )
+
+    # TODO : adding crop = (0,0) for now.
+    # If we do random crop, we have to do this in mediapipe
+    def attach_metadata(batch):
+        import imagesize
+
+        return {
+            "original_sizes": imagesize.get(batch["image"]),
+            "crop_top_lefts": (0, 0),
+        }
 
-        def collate_fn(examples):
-            model_input = torch.stack([torch.tensor(example["model_input"]) for example in examples])
-            original_sizes = [example["original_sizes"] for example in examples]
-            crop_top_lefts = [example["crop_top_lefts"] for example in examples]
-            prompt_embeds = torch.stack([torch.tensor(example["prompt_embeds"]) for example in examples])
-            pooled_prompt_embeds = torch.stack([torch.tensor(example["pooled_prompt_embeds"]) for example in examples])
-
-            return {
-                "model_input": model_input,
-                "prompt_embeds": prompt_embeds,
-                "pooled_prompt_embeds": pooled_prompt_embeds,
-                "original_sizes": original_sizes,
-                "crop_top_lefts": crop_top_lefts,
-            }
+    with accelerator.main_process_first():
+        from datasets.fingerprint import Hasher
+
+        # fingerprint used by the cache for the other processes to load the result
+        # details: https://github.com/huggingface/diffusers/pull/4038#discussion_r1266078401
+        new_fingerprint = Hasher.hash(args)
```

Comment:
right, the reason to move VAE to device is because it's slow on cpu. Also, the original code was using cache that  is not right as our images are transformed.